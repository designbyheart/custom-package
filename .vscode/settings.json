{
  "flow.useNPMPackagedFlow": true,
  "javascript.validate.enable": false,
  "cSpell.words": [
    "Onfido"
  ],
  "flow.useLSP": false,
  "flow.pathToFlow": "${workspaceFolder}/node_modules/.bin/flow",
  "flow.useBundledFlow": false
}