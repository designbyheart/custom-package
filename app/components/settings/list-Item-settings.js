// @flow
import React, { PureComponent } from 'react'
import { StyleSheet, ScrollView, View, Image } from 'react-native'
import { ListItem } from 'react-native-elements'
import { font } from '../../common/styles/constant'

class ListItemSettings extends PureComponent<any, any> {
  render() {
    return (
      <ScrollView bounces={false}>
        <View style={style.listContainer}>
          {this.props.list.map((item, index) => {
            return (
              <ListItem
                containerStyle={[style.listItemContainer]}
                titleStyle={[style.titleStyle]}
                subtitleStyle={[style.subtitleStyle]}
                key={index}
                title={item.title}
                subtitle={item.subtitle}
                leftAvatar={item.avatar}
                rightIcon={
                  item.rightIcon === 'spinner' ? (
                    <View>
                      <Image
                        style={style.spinner}
                        source={require('../../images/loadingSpinner.gif')}
                      />
                    </View>
                  ) : (
                    item.rightIcon
                  )
                }
                // hideChevron={item.rightIcon === 'spinner'}
                onPress={item.onPress}
              />
            )
          })}
        </View>
      </ScrollView>
    )
  }
}
export { ListItemSettings }

const style = StyleSheet.create({
  listContainer: {
    flex: 1,
    justifyContent: 'flex-start',
    borderBottomWidth: 0,
    borderTopWidth: 0,
    backgroundColor: '#fff',
    marginTop: 0,
    paddingBottom: 60,
  },
  listItemContainer: {
    borderBottomWidth: 1,
    borderTopWidth: 0,
    borderBottomColor: '#f2f2f2',
    minHeight: 64,
    justifyContent: 'center',
    paddingTop: 0,
    paddingBottom: 0,
    paddingLeft: 0,
    paddingRight: 15,
  },
  titleStyle: {
    fontFamily: 'Lato',
    fontSize: 17,
    color: '#505050',
    fontWeight: '500',
  },
  subtitleStyle: {
    fontFamily: 'Lato',
    fontWeight: '500',
    fontSize: 11,
    color: '#777',
  },
  spinner: {
    width: 60,
    height: 60,
    marginRight: -10,
  },
})
