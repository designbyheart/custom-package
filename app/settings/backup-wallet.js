// @flow

import React, { PureComponent } from 'react'
import { View, Share } from 'react-native'
import RNFetchBlob from 'react-native-fetch-blob'

import type { BackupWalletProps } from './type-settings'
import { getZippedWalletBackupPath } from '../bridge/react-native-cxs/RNCxs'

export default class BackupWallet extends PureComponent<
  BackupWalletProps,
  void
> {
  static defaultProps = {
    status: '',
  }
  backupWallet = async () => {
    // DUMMY DATA UNTIL MOVED INTO SAGA
    const agencyConfig = {
      agencyUrl: 'something',
      agencyDID: 'something',
      agencyVerificationKey: 'something',
      poolConfig: 'something',
    }
    const documentDirectory = RNFetchBlob.fs.dirs.DocumentDir
    const backup = await getZippedWalletBackupPath(
      documentDirectory,
      agencyConfig
    )

    // TODO: move to backupWalletSaga & figure out flow type check hack for url
    //   Share.share(
    //     {
    //       url: backup,
    //       title: 'Share Your Data Wallet',
    //     },
    //     {
    //       // Android
    //       dialogTitle: 'Share Your Data Wallet',
    //     }
    //   )
    console.log(backup)
  }

  render() {
    const { render, status }: BackupWalletProps = this.props

    return <View>{render(status, this.backupWallet)}</View>
  }
}
