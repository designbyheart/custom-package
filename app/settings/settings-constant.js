export const EDIT_ICON_DIMENSIONS = 22
export const PASS_CODE_ASTERISK_TOP_OFFSET = 5
export const USERNAME_TEST_ID = 'settings-user-name-label'
export const PASS_CODE_TEST_ID = 'settings-pass-code-label'
export const PASS_CODE_ASTERISK_TEST_ID = 'settings-pass-code-asterisk-label'
export const TOUCH_ID_TEST_ID = 'settings-touch-id-label'
