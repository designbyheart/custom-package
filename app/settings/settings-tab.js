// @flow
import React, { Component } from 'react'
import { View } from 'react-native'
import { connect } from 'react-redux'
import SvgCustomIcon from '../components/svg-custom-icon'
import { color } from '../common/styles'

import type { Store } from '../store/type-store'
import { styles } from '../navigator'

class SettingsTab extends Component<any> {
  render() {
    return (
      <View style={styles.icon}>
        <SvgCustomIcon
          name="Settings"
          height={29}
          fill={
            this.props.focused ? color.actions.font.tenth : color.actions.sixth
          }
        />
        {this.props.cloudBackupError && (
          <View
            style={{
              position: 'absolute',
              width: 10,
              height: 10,
              right: 0,
              borderRadius: 50,
              backgroundColor: 'red',
              zIndex: 10002,
            }}
          />
        )}
      </View>
    )
  }
}

const mapStateToProps = (state: Store) => {
  return {
    cloudBackupError: state.backup.cloudBackupError,
  }
}

export default connect(mapStateToProps)(SettingsTab)
