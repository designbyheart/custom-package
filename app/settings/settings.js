import React, { PureComponent } from 'react'
import {
  View,
  Image,
  Text,
  Switch,
  StyleSheet,
  ScrollView,
  StatusBar,
} from 'react-native'
import { StackNavigator } from 'react-navigation'
import { Avatar, CustomText } from '../components'
import { CustomList, CustomView } from '../components/layout'
import { settingsRoute } from '../common/route-constants'
import {
  white,
  mantis,
  LIST_OFFSET_1X,
  barStyleDark,
  color,
} from '../common/styles/constant'
import {
  EDIT_ICON_DIMENSIONS,
  PASS_CODE_ASTERISK_TOP_OFFSET,
  HISTORY_TEST_ID,
  PASS_CODE_ASTERISK_TEST_ID,
  PASS_CODE_TEST_ID,
  TOUCH_ID_TEST_ID,
  USERNAME_TEST_ID,
} from './settings-constant'

const style = StyleSheet.create({
  container: {
    backgroundColor: color.bg.fifth.color,
  },
  headerStyle: {
    backgroundColor: color.bg.fifth.color,
    shadowOpacity: 0,
  },
  editIcon: {
    width: EDIT_ICON_DIMENSIONS,
    height: EDIT_ICON_DIMENSIONS,
  },
  labelImage: {
    marginRight: LIST_OFFSET_1X,
  },
  labelPassCode: {
    marginTop: PASS_CODE_ASTERISK_TOP_OFFSET,
  },
})

const userAvatar = (
  <Avatar medium round src={require('./images/UserAvatar.png')} />
)
const editIcon = (
  <Image
    style={style.editIcon}
    resizeMode={'contain'}
    source={require('./images/edit.png')}
  />
)

const SettingText = props => (
  <CustomText bg="fifth" h4 semiBold {...props}>
    {props.children}
  </CustomText>
)

const userName = <SettingText testID={USERNAME_TEST_ID}>Your Name</SettingText>
const passCode = (
  <CustomView row>
    <SettingText testID={PASS_CODE_TEST_ID}>Passcode: </SettingText>
    <SettingText
      testID={PASS_CODE_ASTERISK_TEST_ID}
      style={[style.labelPassCode]}
    >
      *******
    </SettingText>
  </CustomView>
)
const touchId = (
  <CustomView row>
    <Image
      style={[style.labelImage, style.editIcon]}
      source={require('./images/biometrics.png')}
    />
    <SettingText testID={TOUCH_ID_TEST_ID}>Enable Touch ID</SettingText>
  </CustomView>
)
const toggleSwitch = (
  <Switch onTintColor={mantis} tintColor={white} value={true} />
)
const history = (
  <CustomView row>
    <Image
      style={[style.labelImage, style.editIcon]}
      source={require('./images/history.png')}
    />
    <SettingText testID={HISTORY_TEST_ID}>View History</SettingText>
  </CustomView>
)

const itemList = [
  {
    id: 0,
    left: userAvatar,
    right: editIcon,
  },
  {
    id: 1,
    left: userName,
    right: editIcon,
  },
  {
    id: 2,
    left: passCode,
    right: editIcon,
  },
  {
    id: 3,
    left: touchId,
    right: toggleSwitch,
  },
  {
    id: 4,
    left: history,
  },
]

export class Settings extends PureComponent {
  static navigationOptions = {
    headerTitle: <SettingText>{settingsRoute}</SettingText>,
    headerStyle: style.headerStyle,
  }

  render() {
    return (
      <ScrollView style={style.container}>
        <StatusBar barStyle={barStyleDark} />
        <CustomList
          data={itemList}
          listStyle={style.list}
          itemStyle={style.item}
        />
      </ScrollView>
    )
  }
}

export default StackNavigator({
  [settingsRoute]: {
    screen: Settings,
  },
})
