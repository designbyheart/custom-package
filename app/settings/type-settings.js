// @flow
import type { ReactNavigation } from '../common/type-common'

export type SettingsProps = {
  touchIdActive: boolean,
  selectUserAvatar: () => void,
} & ReactNavigation
