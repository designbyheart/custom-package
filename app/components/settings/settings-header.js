// @flow
import React, { Component } from 'react'
import {
  StyleSheet,
  View,
  Text,
  TouchableOpacity,
  SafeAreaView,
} from 'react-native'
import { SvgSettingIcons } from '../../components/svg-setting-icons'
import { UserAvatar, Avatar } from '../../components'
import { USER_AVATAR_TEST_ID } from '../../settings/settings-constant'
import type { ImageSource } from '../../common/type-common'
import { measurements } from '../../common/styles/measurements'

class SettingsHeader extends React.Component<any, any> {
  renderAvatarWithSource = (avatarSource: number | ImageSource) => (
    <Avatar medium round src={avatarSource} />
  )
  render() {
    return (
      <View style={style.container}>
        <SafeAreaView style={style.safeView}>
          <UserAvatar testID={USER_AVATAR_TEST_ID} userCanChange>
            {this.renderAvatarWithSource}
          </UserAvatar>
          <Text style={style.userName}>Me</Text>
          {/* <View style={style.tokensWrapper}>
            <TouchableOpacity onPress={this.props.tokenScreen}>
              <SvgSettingIcons
                name="TokenIcon"
                width="20"
                height="20"
                fill="#EB9B2D"
              />
            </TouchableOpacity>
            <TouchableOpacity onPress={this.props.tokenScreen}>
              <Text style={style.tokens}>{this.props.balance}</Text>
            </TouchableOpacity>
          </View> */}
          {/* <Text style={style.tokensLabel}>TOKENS</Text> */}
        </SafeAreaView>
      </View>
    )
  }
}
export { SettingsHeader }

const style = StyleSheet.create({
  container: {
    height: measurements.settingsHeader,
    alignItems: 'center',
    justifyContent: 'center',
    shadowColor: '#000000',
    shadowOpacity: 0.3,
    shadowRadius: 12,
    elevation: 12,
    position: 'relative',
    zIndex: 100,
    backgroundColor: '#fff',
    borderTopWidth: 0,
    borderLeftWidth: 0,
    borderRightWidth: 0,
    borderBottomWidth: 1,
    borderBottomColor: 'transparent',
  },
  safeView: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  avatarWrapper: {
    width: 64,
    height: 64,
    borderRadius: 32,
    overflow: 'hidden',
    backgroundColor: '#eaeaea',
    alignItems: 'center',
    justifyContent: 'flex-end',
  },
  userName: {
    fontFamily: 'Lato',
    fontSize: 17,
    fontWeight: '700',
    color: '#505050',
    marginTop: 12,
    marginBottom: 11,
  },
  tokensWrapper: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
  tokens: {
    fontFamily: 'Lato',
    fontSize: 22,
    fontWeight: '400',
    color: '#505050',
    marginLeft: 10,
  },
  tokensLabel: {
    fontFamily: 'Lato',
    fontSize: 11,
    fontWeight: '500',
    color: '#777',
  },
})
