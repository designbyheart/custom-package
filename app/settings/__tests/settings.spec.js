import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import { Settings } from '../settings'

describe('user settings screen', () => {
  xit('should render properly and snapshot should match', () => {
    const component = renderer.create(<Settings />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
