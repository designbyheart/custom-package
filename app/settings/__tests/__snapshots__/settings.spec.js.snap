// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user settings screen should render properly and snapshot should match 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#f0f0f0",
      "flex": 1,
    }
  }
  tertiary={true}
>
  <View
    style={
      Object {
        "overflow": "scroll",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "marginHorizontal": 20,
          },
          undefined,
        ]
      }
    >
      <View
        row={true}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "marginVertical": 20,
          }
        }
        vCenter={true}
      >
        <View
          left={true}
          style={
            Object {
              "alignItems": "flex-start",
              "flex": 1,
            }
          }
        >
          <View
            imageShadow={null}
            style={
              Object {
                "borderRadius": 38,
              }
            }
          >
            <Image
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hitSlop={undefined}
              onError={undefined}
              onLayout={undefined}
              onLoad={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              source={1}
              style={
                Array [
                  Object {
                    "height": 76,
                    "width": 76,
                  },
                  Object {
                    "borderRadius": 38,
                  },
                ]
              }
              testID={undefined}
            />
          </View>
        </View>
        <View
          right={true}
          style={
            Object {
              "alignItems": "flex-end",
            }
          }
        >
          <Image
            resizeMode="contain"
            source={1}
            style={
              Array [
                Object {
                  "height": 30,
                  "width": 30,
                },
                Object {
                  "tintColor": "grey",
                },
              ]
            }
          />
        </View>
      </View>
      <View
        row={true}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "marginVertical": 20,
          }
        }
        vCenter={true}
      >
        <View
          left={true}
          style={
            Object {
              "alignItems": "flex-start",
              "flex": 1,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontSize": 17,
                },
                Object {
                  "fontFamily": "SF Pro Display",
                },
                Object {
                  "backgroundColor": "#f0f0f0",
                  "color": "#777777",
                },
                Object {
                  "fontWeight": "600",
                },
                null,
                null,
              ]
            }
            testID="settings-user-name-label"
          >
            Name: Anonymous
          </Text>
        </View>
        <View
          right={true}
          style={
            Object {
              "alignItems": "flex-end",
            }
          }
        >
          <Image
            resizeMode="contain"
            source={1}
            style={
              Array [
                Object {
                  "height": 30,
                  "width": 30,
                },
                Object {
                  "tintColor": "grey",
                },
              ]
            }
          />
        </View>
      </View>
      <View
        row={true}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "marginVertical": 20,
          }
        }
        vCenter={true}
      >
        <View
          left={true}
          style={
            Object {
              "alignItems": "flex-start",
              "flex": 1,
            }
          }
        >
          <View
            row={true}
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "fontSize": 17,
                  },
                  Object {
                    "fontFamily": "SF Pro Display",
                  },
                  Object {
                    "backgroundColor": "#f0f0f0",
                    "color": "#777777",
                  },
                  Object {
                    "fontWeight": "600",
                  },
                  null,
                  null,
                ]
              }
              testID="settings-pass-code-label"
            >
              Pass code:
               
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "fontSize": 17,
                  },
                  Object {
                    "fontFamily": "SF Pro Display",
                  },
                  Object {
                    "backgroundColor": "#f0f0f0",
                    "color": "#777777",
                  },
                  Object {
                    "fontWeight": "600",
                  },
                  null,
                  null,
                  Object {
                    "marginTop": 5,
                  },
                ]
              }
              testID="settings-pass-code-asterisk-label"
            >
              ******
            </Text>
          </View>
        </View>
        <View
          right={true}
          style={
            Object {
              "alignItems": "flex-end",
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={
              Object {
                "bottom": 15,
                "left": 15,
                "right": 15,
                "top": 15,
              }
            }
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Image
              resizeMode="contain"
              source={1}
              style={
                Array [
                  Object {
                    "height": 30,
                    "width": 30,
                  },
                  Object {
                    "tintColor": "grey",
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
      <View
        row={true}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "marginVertical": 20,
          }
        }
        vCenter={true}
      >
        <View
          left={true}
          style={
            Object {
              "alignItems": "flex-start",
              "flex": 1,
            }
          }
        >
          <View
            row={true}
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Image
              source={1}
              style={
                Array [
                  Object {
                    "marginRight": 10,
                  },
                  Object {
                    "height": 30,
                    "width": 30,
                  },
                  Object {
                    "tintColor": "#85bf43",
                  },
                ]
              }
            />
            <View
              center={true}
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                onPress={[Function]}
                style={
                  Array [
                    Object {
                      "fontSize": 17,
                    },
                    Object {
                      "fontFamily": "SF Pro Display",
                    },
                    Object {
                      "backgroundColor": "#f0f0f0",
                      "color": "#777777",
                    },
                    Object {
                      "fontWeight": "600",
                    },
                    null,
                    null,
                  ]
                }
                testID="settings-touch-id-label"
              >
                Enable Touch ID
              </Text>
            </View>
          </View>
        </View>
        <View
          right={true}
          style={
            Object {
              "alignItems": "flex-end",
            }
          }
        >
          <RCTSwitch
            disabled={false}
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onTintColor="#85bf43"
            onValueChange={[Function]}
            style={
              Array [
                Object {
                  "height": 31,
                  "width": 51,
                },
                undefined,
              ]
            }
            tintColor="#ffffff"
            value={false}
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
